azuredevops/internal/service/graph/data_group.go

```diff
@@ -19,15 +19,14 @@ func DataGroup() *schema.Resource {
 		Schema: map[string]*schema.Schema{
 			"name": {
 				Type:         schema.TypeString,
-				ForceNew:     true,
 				Required:     true,
-				ValidateFunc: validation.NoZeroValues,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
 			},
 			"project_id": {
 				Type:         schema.TypeString,
-				ForceNew:     true,
-				Required:     true,
-				ValidateFunc: validation.NoZeroValues,
+				Optional:     true,
+				Computed:     true,
```

Comment:
`project_id` is required when we need to read project groups.  Both of  project groups and collection groups won't set `project_id` when service response returned. I think  we can remove `Computed:     true,` as this configuration is configured by the users.

Reply:
Yes I agree, you're right. I removed the `computed` flag for `project_id`